### PR TITLE
Add RobotPy setup to zero-to-robot

### PR DESCRIPTION
**Pull request by virtuald**
I have one more section to push that goes along with this, but we can't merge until the robotpy meta package is pushed later tonight.

### CODE REVIEW

File: source/docs/zero-to-robot/step-2/python-setup.rst [hunk 1]
```diff
@@ -0,0 +1,227 @@
...
+      If you don't have administrative rights on your computer, either use `virtualenv/virtualenvwrapper <http://docs.python-guide.org/en/latest/dev/virtualenvs/>`_, or you can install to the user site-packages directory:
+
+      .. code-block:: sh
+
+         pip3 install --user robotpy
+
+   .. tab-item:: Linux
+      :sync: linux
+
+      RobotPy distributes manylinux binary wheels on PyPI. However, installing these requires a distro that has glibc 2.35 or newer, and an installer that implements :pep:`600`, such as pip 20.3 or newer. You can check your version of pip with the following command:
```
Comment (sciencewhiz) [inline source/docs/zero-to-robot/step-2/python-setup.rst:75]: Up above it says glibc 2.34 or newer

File: source/docs/zero-to-robot/step-2/python-setup.rst [hunk 2]
```diff
@@ -0,0 +1,227 @@
...
+
+      To upgrade, you can run this:
+
+      .. code-block:: sh
+
+         pip3 install --upgrade robotpy
+
+      The following Linux distributions are known to work, but this list is not necessarily comprehensive:
+
+      * Ubuntu 22.04+
```
Comment (sciencewhiz) [inline source/docs/zero-to-robot/step-2/python-setup.rst:97]: Should this go above with supported languages?

File: source/docs/zero-to-robot/step-2/python-setup.rst (hunk 1 above)
Comment (virtuald) [reply source/docs/zero-to-robot/step-2/python-setup.rst:75]: Lame.

File: source/docs/zero-to-robot/step-2/python-setup.rst [hunk 3]
```diff
@@ -0,0 +1,221 @@
...
+
+      **source install**
+
+      Alternatively, if you have a C++20 compiler installed, you may be able to use pip to install RobotPy from source.
+
+      .. warning:: It may take a very long time to install!
+
+      .. warning:: Mixing our pre-built wheels with source installs may cause runtime errors. This is due to internal ABI incompatibility between compiler versions.
+
+         Our ARM wheels are built on Ubuntu 22.04 with GCC 10.
```
Comment (auscompgeek) [inline source/docs/zero-to-robot/step-2/python-setup.rst:116]: Don't our ARM builds target Debian bullseye?

Comment (virtuald) [reply source/docs/zero-to-robot/step-2/python-setup.rst:116]: I dunno, just copy/pasting.

Comment (auscompgeek) [reply source/docs/zero-to-robot/step-2/python-setup.rst:116]:
```suggestion
         Our ARM wheels are built for Debian 11 with GCC 10.
```

**Comment by virtuald** [comment]
This is good to go.

File: source/docs/zero-to-robot/step-2/python-setup.rst [hunk 4]
```diff
@@ -0,0 +1,191 @@
...
+         py -3 -m pip install --user robotpy
+
+   .. tab-item:: macOS
+      :sync: macos
+
+      On a macOS system that has pip installed, just run the following command from the Terminal application (may require admin rights):
+
+      .. code-block:: sh
+
+         pip3 install robotpy
```
Comment (TheTripleV) [inline source/docs/zero-to-robot/step-2/python-setup.rst:58]: Use `python3 -m pip` for all these. Below, `python3` is used, which may point to a different python than `pip3`

File: source/docs/software/wpilib-tools/robot-simulation/introduction.rst [hunk 5]
```diff
@@ -31,6 +31,11 @@ Ensure the :guilabel:`Desktop Development with C++` option is checked in the Vis
 .. image:: images/vs-build-tools.png
    :alt: Screenshot of the Visual Studio build tools option
 
+Enabling Desktop Support (Python)
+---------------------------------
+
+GUI simulation support is installed when you ``pip install robotpy``.
```
Comment (TheTripleV) [inline source/docs/software/wpilib-tools/robot-simulation/introduction.rst:37]:
```suggestion
GUI simulation support is installed when you install robotpy.
```

The install instructions use other pip commands. All the install instructions should be isolated to that page

File: source/docs/software/wpilib-tools/robot-simulation/introduction.rst [hunk 6]
```diff
@@ -31,6 +31,11 @@ Ensure the :guilabel:`Desktop Development with C++` option is checked in the Vis
...
+
```
Comment (TheTripleV) [inline source/docs/software/wpilib-tools/robot-simulation/introduction.rst:38]:
```
Running Robot Simulation (Python)
------------------------

Robot simulation can be done using the command line, running `python3 -m robotpy sim`.
```
^ Or, link to the robotpy sim instructions from here.

Google search will probably link to here and not to the other robotpy sim instructions

**Review by TheTripleV — COMMENTED** [review]
Rename the page `Python Installation Guide` to `RobotPy Installation Guide`?

File: source/docs/zero-to-robot/step-2/wpilib-setup.rst [hunk 7]
```diff
@@ -1,7 +1,7 @@
 WPILib Installation Guide
 =========================
 
-This guide is intended for Java and C++ teams. LabVIEW teams can skip to :doc:`labview-setup`. Additionally, the below tutorial shows Windows 10, but the steps are identical for all operating systems. Notes differentiating operating systems will be shown.
+This guide is intended for Java and C++ teams. LabVIEW teams can skip to :doc:`labview-setup`. Python teams can skip to :doc:`python-setup`.Additionally, the below tutorial shows Windows 10, but the steps are identical for all operating systems. Notes differentiating operating systems will be shown.
```
Comment (sciencewhiz) [inline source/docs/zero-to-robot/step-2/wpilib-setup.rst:4]:
```suggestion
This guide is intended for Java and C++ teams. LabVIEW teams can skip to :doc:`labview-setup`. Python teams can skip to :doc:`python-setup`. Additionally, the below tutorial shows Windows 10, but the steps are identical for all operating systems. Notes differentiating operating systems will be shown.
```

File: source/docs/software/vscode-overview/3rd-party-libraries.rst [hunk 8]
```diff
@@ -34,6 +34,14 @@ For LabVIEW teams, there might be a few new :guilabel:`Third Party` items on var
 
 In order to install third party libraries for LabVIEW, download the VIs from the vendor (typically via some sort of installer). Then drag and drop the third party VIs into the respective folder mentioned above just like any other VI.
 
+How Does It Work? - Python
```
Comment (sciencewhiz) [inline source/docs/software/vscode-overview/3rd-party-libraries.rst:37]: Down below in the list of vendor deps, we should probably add references to the python package

File: source/docs/software/python/subcommands/index.rst [hunk 9]
```diff
@@ -0,0 +1,62 @@
...
+         python3 -m robotpy sim --help
+
+   .. tab-item:: Linux
+      :sync: linux
+
+      .. code-block:: sh
+
+         python3 -m robotpy sim --help
+
+This documentation site has more detailed documentation for some of the subcommands:
```
Comment (sciencewhiz) [inline source/docs/software/python/subcommands/index.rst:56]:
```suggestion
This page has more detailed documentation for some of the subcommands:
```

File: source/docs/software/python/index.rst [hunk 10]
```diff
@@ -0,0 +1,11 @@
...
+======================
+
+.. toctree::
+   :maxdepth: 2
+   :titlesonly:
+
+   /docs/zero-to-robot/step-2/python-setup
+   /docs/zero-to-robot/step-2/frc-game-tools
+   pyproject_toml
+   subcommands/index
```
Comment (sciencewhiz) [inline source/docs/software/python/index.rst:11]: Is this how you wanted it formatted?
image

**Review by sciencewhiz — COMMENTED** [review]
It would probably be helpful to have a few notes where things are different then previous years for returning teams

File: source/docs/software/python/index.rst (hunk 10 above)
Comment (virtuald) [reply source/docs/software/python/index.rst:11]: My thought was that in the future someone would document some of the other subcommands

File: source/docs/software/vscode-overview/3rd-party-libraries.rst (hunk 8 above)
Comment (virtuald) [reply source/docs/software/vscode-overview/3rd-party-libraries.rst:37]: I agree that we should add references to the python packages, but it's likely that we could bikeshed it quite a bit. Let's do it in a future PR.

**Comment by virtuald** [comment]
Review comments addressed.

**Review by sciencewhiz — APPROVED** [review]

**Pull request merged**